src/App.jsx

```diff
@@ -14,33 +16,110 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+const ENTER_GOODS = goodsFromServer;
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function prepareSortedGoods(goods, sortField, reverse) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [firstList, nextList] = useState(goodsFromServer);
+
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = prepareSortedGoods(firstList, sortField, reverse);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => {
+            nextList(ENTER_GOODS);
+            setSortField(SORT_FIELD_NAME);
+          }}
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_FIELD_NAME,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            nextList(ENTER_GOODS);
+            setSortField(SORT_FIELD_LENGTH);
+          }}
+          className={cn('button is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            setSortField(sortField);
```

Comment:
The line 'setSortField(sortField);' seems unnecessary here as it sets the state to its current value. Consider removing it to avoid redundant operations.